### PR TITLE
Expand subqueries in VALUES statements.

### LINKED ISSUE #2621: sql: expand subqueries in `INSERT` statements
We currently only expand subqueries in `scanNode` and `planner.Update`; we should also expand them in the `VALUES` part of an `INSERT` statement. Motivating example query:

`INSERT INTO tbl (a, b, c) VALUES ($1, $2, (SELECT COALESCE(MAX(c), 0) FROM tbl)+1);`

More generally, subqueries returning exactly one column and one row should be probably be allowed in nearly any expression.

### PR DESCRIPTION
**Pull request by petermattis**
Fixes #2621.

### CODE REVIEW

File: sql/testdata/subquery [hunk 1]
```diff
@@ -61,3 +61,39 @@ query I
...
 ----
 NULL
+
+query I
+VALUES (1, (SELECT (2)))
+----
+1 2
+
+statement ok
+INSERT INTO abc VALUES ((SELECT 7), (SELECT 8), (SELECT 9))
```
Comment (tamird) [inline sql/testdata/subquery]: I think there are two more test cases we need here:
- `INSERT INTO abc VALUES ((SELECT (7,8,9)))`; in postgres, this doesn't type check
- `INSERT INTO abc VALUES ((SELECT 7,8,9))`; in postgres, this errors out with `subquery must return only one column`

Can you add `

Comment (petermattis) [inline sql/testdata/subquery]: Yes. And there are corresponding cases for `UPDATE`. These found bugs in insert/update which are now fixed.

File: sql/insert.go [hunk 2]
```diff
@@ -124,6 +126,15 @@ func (p *planner) Insert(n *parser.Insert) (planNode, error) {
 			}
 		}
 
+		// Check that the row value types match the column types.
```
Comment (tamird) [inline sql/insert.go]: comment about this having to happen before encoding

File: sql/testdata/subquery (hunk 1 above)
Comment (tamird) [inline sql/testdata/subquery]: Nice.

**Comment by tamird** [comment]
LGTM

File: sql/insert.go (hunk 2 above)
Comment (petermattis) [inline sql/insert.go]: Done.

**Comment by vivekmenezes** [comment]
Nice!

**Pull request merged**